### PR TITLE
feat: add fedimintd tagged version endpoint

### PR DESCRIPTION
**Pull request by bradleystachurski**
Closes https://github.com/fedimint/fedimint/issues/5866

Please refer to the associated issue for the motivation to add this endpoint.

```bash
bash-5.2$ fedimint-cli dev peer-version --peer-id 0
{
  "version": "0.5.0-alpha"
}
```

### CODE REVIEW

**Review by m1sterc001guy — DISMISSED** [review]
Looks good to me. I assume we don't want to make this authenticated so that Fedimint-observer can use it?

File: fedimint-cli/src/client.rs [hunk 1]
```diff
@@ -195,6 +195,11 @@ pub enum ClientCmd {
     Config,
     /// Gets the current fedimint AlephBFT session count
     SessionCount,
+    /// Gets the tagged fedimintd version for a peer
+    FedimintdVersion {
```
Comment (dpc) [inline fedimint-cli/src/client.rs:199]: Maybe this should be under `dev`, and maybe named `fedimint-cli dev peer-version`?

**Review by dpc — COMMENTED** [review]
LGTM otherwise

**Comment by bradleystachurski** [comment]
> I assume we don't want to make this authenticated so that Fedimint-observer can use it?

@m1sterc001guy yep, that's exactly the reason :+1:

**Comment by fedimint-backports** [comment]
Backport failed for `releases/v0.4`, because it was unable to cherry-pick the commit(s).

Please cherry-pick the changes locally and resolve any conflicts.
```bash
git fetch origin releases/v0.4
git worktree add -d .worktree/backport-5973-to-releases/v0.4 origin/releases/v0.4
cd .worktree/backport-5973-to-releases/v0.4
git switch --create backport-5973-to-releases/v0.4
git cherry-pick -x b9d8ba7cf856c556c466e2fedc3fddc76e38ac3e
```

**Pull request merged**